api/src/test/java/org/openmrs/module/locationbasedaccess/aop/LocationUtilsTest.java

```diff
@@ -0,0 +1,114 @@
+package org.openmrs.module.locationbasedaccess.aop;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import org.openmrs.Location;
+import org.openmrs.Person;
+import org.openmrs.PersonAttributeType;
+import org.openmrs.User;
+import org.openmrs.api.LocationService;
+import org.openmrs.api.PersonService;
+import org.openmrs.api.UserService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.locationbasedaccess.LocationBasedAccessConstants;
+import org.openmrs.module.locationbasedaccess.utils.LocationUtils;
+import org.openmrs.test.BaseModuleContextSensitiveTest;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public class LocationUtilsTest extends BaseModuleContextSensitiveTest {
+
+	private PersonService personService;
+	private LocationService locationService;
+	private UserService userService;
+	private static final String XML_FILENAME_WITH_PERSON_DATA = "include/LocationUtilTestData.xml";
+	private static final int DEMO_PERSON3_ID = 3;
+	private static final int DEMO_PERSON2_ID = 2;
+	private static final int DEMO_PERSON4_ID = 4;
+	private static final int DEMO_PERSON5_ID = 6005;
+	private static final int DEMO_PERSON6_ID = 6006;
+	private static final int DEMO_LOCATION1_ID = 1;
+	private static final int DEMO_USER1_ID = 6001;
+	private static final int DEMO_USER2_ID = 6002;
+	private static final int DEMO_USER3_ID = 6003;
+
+	public LocationUtilsTest(){
+
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		userService = Context.getUserService();
+		personService = Context.getPersonService();
+		locationService = Context.getLocationService();
+		executeDataSet(XML_FILENAME_WITH_PERSON_DATA);
+	}
+
+	@Test
+	public void getPersonLocation_shouldReturnPersonLocation(){
+		Person person3 = personService.getPerson(DEMO_PERSON3_ID);
+		Location location = locationService.getLocation(DEMO_LOCATION1_ID);
+		Location location3 = LocationUtils.getPersonLocation(person3);
+		assertEquals(location,location3);
```

Comment:
How can you ensure that, if the LocationUtils.getPersonLocation() returns the same location as locationService.getLocation()?

Reply:
Here in the Dataset person with Id `DEMO_PERSON3_ID` is associated with the location having an ID  `DEMO_LOCATION1_ID`.

Reply:
> Dataset person with Id DEMO_PERSON3_ID is associated with the location having an ID DEMO_LOCATION1_ID

Here, How can you show it through the code? 
For a unit test, we should not attach any human justifications for the results. Here you trust your Dataset and assuming that the person has that one.

A good way would be, 
1. Create a location - LocationA
2. Create a person
3. Add location to that person
4. Get the location from the person through our method. 
5. Now check for the response against LocationA
All of the above methods should be implemented through the codes.

Reply:
Follow it for other test cases as well !!, You can get anything from Dataset, but you need to add test to ensure that you are getting same from the dataset